lua/nvim-treesitter/locals.lua

```diff
@@ -2,20 +2,35 @@
 -- Locals are a generalization of definition and scopes
 -- its the way nvim-treesitter uses to "understand" the code
 local api = vim.api
-local ts = vim.treesitter
 
 local queries = require'nvim-treesitter.query'
 local parsers = require'nvim-treesitter.parsers'
 
-local M = {
-  locals = {}
+local default_dict = {
+  __index = function(table, key)
+    local exists = rawget(table, key)
+    if not exists then
+      table[key] = {}
+    end
+    return rawget(table, key)
+  end
 }
 
-function M.collect_locals(bufnr)
+local query_chache
+if not query_chache then
+  query_chache = {}
+  setmetatable(query_chache, default_dict)
+end
```

Comment:
i think you meant `cache` ?

Reply:
Query_cache should be the default_dict to support arbitrary query kinds. Query_cachd holds ticks and cache for each query kind.  Maybe you know a better name? I initialize with {} whenever a new local type is requested for the first time

Reply:
i meant `cache` not `chache`